### DISCUSSION TITLE
Failure when checking if mesh-id is configured: Unauthorized

### DISCUSSION THREAD

**Discussion by owen200008** [kiali/kiali#5462] [General]
when use the istio install https://istio.io/latest/docs/setup/getting-started/
version：1.15.0
the web show like ： Kiali failed to initialize.Please ensure that services Kiali depends on, such as Prometheus, are healthy and reachable by Kiali then refresh your browser.

pod only find the log Failure when checking if mesh-id is configured: Unauthorized

how to resolve？

**Comment by jshaughn** [comment]
I've converted this to a discussion while we try to figure out your issue.  We'll create a bug if one is identified.

Did you install all of the Istio add-ons, especially Prometheus, and can you confirm that those pods are running?

**Comment by owen200008** [comment]
yes, all is running status
cni use cilium 
in the kiala pod，maybe can not get info from api-server

**Comment by owen200008** [comment]
yes, all is running status
cni use cilium 
in the kiali pod，maybe can not get info from api-server

**Reply by owen200008** [reply]
ectd, api-server, controller install in master node, not use pod, use service point to api-server

**Reply by owen200008** [reply]
another k8s use cilium-istioctl 1.10.6 is success.
k8s install same way (api-server is not pod)

**Comment by israel-hdez** [comment]
Kiali installation may be bad, because it may not have access to the Istio ConfigMap.
How did you install Kiali?

**Reply by owen200008** [reply]
istio 1.15.0 apply -f samples/addin

**Reply by israel-hdez** [reply]
What's your kubernetes version?

**Reply by jmazzitelli** [reply]
That's the simple kiali addon install, so nothing special going on there.

It sounds like you simply have a cluster configuration where your cluster has prohibited connectivity between the kiali pod and the prometheus pod (or the cluster API server). Look at your cluster configuration and make sure it allows connectivity between pods like that. Kiali needs access to the prometheus pod as well as access to the cluster api server.

You can try a simple `curl` command from the kiali pod to the prometheus pod - my guess is that it will fail. Get that to work, and you will solve the mystery.

Try something like this:

```
kubectl exec -n istio-system deploy/kiali -- curl http://prometheus.istio-system.svc:9090 -s
```

That needs to succeed. Kiali needs access to the Prometheus service. When I run it I see:

```
$ kubectl exec -n istio-system deploy/kiali -- curl http://prometheus.istio-system.svc:9090 -s
<a href="/graph">Found</a>.
```

(the `<a href="/graph">Found</a>` is just the prometheus server doing a redirect to the /graph page - if you see that, you know you can hit the prometheus server)

**Reply by owen200008** [reply]
curl http://prometheus.istio-system.svc:9090 is success
i add this like follow, and reset coredns(i think not him), kiali success.

apiVersion: v1
kind: ConfigMap
metadata:
  name: ip-masq-agent
  namespace: kube-system
data:
  config: "nonMasqueradeCIDRs:\n  - 172.16.0.0/16 \nmasqLinkLocal: false\nmasqLinkLocalIPv6:
    false\nresyncInterval: 60s\n\n"

**Reply by israel-hdez** [reply]
Please, check that the Kiali service account got the right privileges. It looks like it cannot fetch Istio's ConfigMap.
